sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -197,7 +197,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * @since 1.4.0
    */
   def jdbc(url: String, table: String, properties: Properties): DataFrame = {
-    assertNoSpecifiedSchema("jdbc")
+    assertJdbcAPISpecifiedDataFrameSchema()
```

Comment:
Users should be able to do it in either way. If users specify them in both `schema()` API and the `customerSchema` option, we should issue an exception.